app/src/main/java/org/mozilla/fenix/compose/Chip.kt

```diff
@@ -0,0 +1,196 @@
+    backgroundColor: Color = FirefoxTheme.colors.actionPrimary,
+    onClick: () -> Unit,
+) {
+    androidx.compose.material.Chip(
+        onClick = onClick,
+        modifier = modifier,
+        colors = chipColors(
+            backgroundColor = backgroundColor,
+            contentColor = textColor,
+        ),
+    ) {
+        Text(
+            text = text,
+            color = textColor,
+            style = FirefoxTheme.typography.body2,
+        )
+    }
+}
+
+/**
+ * Default layout of a selectable chip.
+ *
+ * @param text [String] displayed in this chip.
+ * @param isSelected Whether this should be shown as selected.
+ * @param selectableChipColors The color set defined by [SelectableChipColors] used to style the chip.
+ * @param onClick Callback for when the user taps this.
+ */
+@Composable
+fun SelectableChip(
+    text: String,
+    isSelected: Boolean,
+    selectableChipColors: SelectableChipColors = SelectableChipColors.buildColors(),
+    onClick: () -> Unit,
+) {
+    var selected by remember { mutableStateOf(isSelected) }
+
+    Chip(
+        text = text,
+        textColor = if (selected) selectableChipColors.selectedTextColor else selectableChipColors.unselectedTextColor,
+        backgroundColor = if (selected) {
+            selectableChipColors.selectedBackgroundColor
+        } else {
+            selectableChipColors.unselectedBackgroundColor
+        },
+    ) {
+        selected = !selected
+        onClick()
+    }
+}
+
+/**
+ * Wrapper for the color parameters of [SelectableChip].
+ *
+ * @param selectedTextColor Text [Color] when the chip is selected.
+ * @param unselectedTextColor Text [Color] when the chip is not selected.
+ * @param selectedBackgroundColor Background [Color] when the chip is selected.
+ * @param unselectedBackgroundColor Background [Color] when the chip is not selected.
+ */
+data class SelectableChipColors(
+    val selectedBackgroundColor: Color,
+    val unselectedBackgroundColor: Color,
+    val selectedTextColor: Color,
+    val unselectedTextColor: Color,
+) {
+    companion object {
+
+        /**
+         * Builder function used to construct an instance of [SelectableChipColors].
+         */
+        @Composable
+        fun buildColors(
+            selectedBackgroundColor: Color = FirefoxTheme.colors.actionPrimary,
+            unselectedBackgroundColor: Color = FirefoxTheme.colors.actionTertiary,
+            selectedTextColor: Color = FirefoxTheme.colors.textActionPrimary,
+            unselectedTextColor: Color = FirefoxTheme.colors.textActionTertiary,
+        ) = SelectableChipColors(
+            selectedBackgroundColor = selectedBackgroundColor,
+            unselectedBackgroundColor = unselectedBackgroundColor,
+            selectedTextColor = selectedTextColor,
+            unselectedTextColor = unselectedTextColor,
+        )
+    }
+}
+
+@Composable
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_YES)
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_NO)
```

Comment:
make sure to replace this with `LightDarkPreview` after that PR lands

Reply:
I'll fix the commits after I include this change after the PR for `LightDarkPreview` lands